apps/rush-lib/src/cli/utilities/InstallManager.ts

```diff
@@ -152,44 +152,49 @@ export default class InstallManager {
       fsx.mkdirSync(rushHomeFolder);
     }
 
-    // Example: "C:\Users\YourName\.rush\npm-1.2.3"
-    const npmToolFolder: string = path.join(rushHomeFolder, 'npm-' + this._rushConfiguration.npmToolVersion);
-    // Example: "C:\Users\YourName\.rush\npm-1.2.3\last-install.flag"
-    const npmToolFlagFile: string = path.join(npmToolFolder, 'last-install.flag');
+    const packageManager: 'pnpm' | 'npm' = this._rushConfiguration.packageManager;
+    const packageManagerVersion: string = this._rushConfiguration.packageManagerToolVersion;
+
+    // Example: "C:\Users\YourName\.rush\pnpm-1.2.3"
+    const packageManagerToolFolder: string = path.join(rushHomeFolder,
+      `${packageManager}-${packageManagerVersion}`);
+    // Example: "C:\Users\YourName\.rush\pnpm-1.2.3\last-install.flag"
+    const packageManagerToolFlagFile: string = path.join(packageManagerToolFolder, 'last-install.flag');
 
     // NOTE: We don't care about the timestamp for last-install.flag, because nobody will change
     // the package.json for this case
-    if (forceReinstall || !fsx.existsSync(npmToolFlagFile)) {
-      console.log(colors.bold('Installing NPM version ' + this._rushConfiguration.npmToolVersion) + os.EOL);
+    if (forceReinstall || !fsx.existsSync(packageManagerToolFlagFile)) {
+      console.log(colors.bold(`Installing ${packageManager} version ${packageManagerVersion}${os.EOL}`));
 
       Utilities.installPackageInDirectory(
-        npmToolFolder,
-        'npm',
-        this._rushConfiguration.npmToolVersion,
-        'npm-local-install',
+        packageManagerToolFolder,
+        packageManager,
+        this._rushConfiguration.packageManagerToolVersion,
+        `${packageManager}-local-install`,
```

Comment:
> `${packageManager}-local-install`, [](start = 7, length = 35)

Make sure the incremental "rush install" behavior works without errors after switching to and from PNPM.